### PR TITLE
doc: Add Thinger.io Cloud Integration

### LINKED ISSUE #951: Add Thinger.io configuration in Cloud Integration section
#### Summary

[Thinger.io](https://thinger.io) implements an integration with The Things Stack. This issue is to propose the addition of this section into The Things Stack Cloud Integrations.

#### Why do we need this ?

Gives knowledge and support to any user integration The Things Stack and Thinger.io together.

#### What is already there? What do you see now?

No documentation as of yet.

#### What is missing? What do you want to see?

...

#### How do you propose to document this?

I will add the necessary information. 

#### Can you do this yourself and submit a Pull Request?

Yes.

### PR DESCRIPTION
**Pull request by bautisflow**
#### Summary

Closes #951

#### Screenshots

tts_rendered

#### Changes

- doc/content/integrations/cloud-integrations/thinger-io/_index.md
- Images under doc/content/integrations/cloud-integrations/thinger-io/

#### Notes for Reviewers

There is an open pull request to integrate this integration into The Things Stack Webhook Templates: https://github.com/TheThingsNetwork/lorawan-webhook-templates/pull/70

Our own documentation is under: https://docs.thinger.io/plugins/the-things-stack

#### Checklist

- [X] Scope: The referenced issue is addressed, there are no unrelated changes.
- [X] Run Locally: Verified that the docs build using `make server`, posted screenshots, verified external links. Test with `HUGO_PARAMS_SEARCH_ENABLED=true` if style changes will affect the search bar.
- [ ] New Features Marked: Documentation for new features is marked using the `new-in-version` shortcode, according to the guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [X] Style Guidelines: Documentation obeys style guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [X] Commits: Commit messages follow guidelines in [CONTRIBUTING](CONTRIBUTING.md), there are no fixup commits left.

### CODE REVIEW

File: doc/content/integrations/cloud-integrations/thinger-io/_index.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
+---
+title: "Thinger.io"
+description: ""
+weight: 
+---
+
+[Thinger.io](https://thinger.io/) is a cloud IoT Platform that provides every needed tool to prototype, scale and manage connected products in a very simple way. Our goal is to democratize the use of IoT making it accessible to the whole world, and streamlining the development of big IoT projects.
+
+From [Thinger.io](https://thinger.io/) we wanted to offer an improved integration to {{% tts %}} users by providing easy to configure tools for the storing, analyzing and showing devices data in a simple way. [The plugin](https://docs.thinger.io/plugins/the-things-stack) allows to retrieve {{% tts %}} webhook messages to enhance the integration with some interesting features such as:
```
Comment (nejraselimovic) [inline doc/content/integrations/cloud-integrations/thinger-io/_index.md:9]: All fine, but can you please rephrase the **we** and **our** in these sentences? It's a bit weird to use **we** for Thinger.io when it's a TTS/TTI webpage

File: doc/content/integrations/cloud-integrations/thinger-io/_index.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+1. Have an active subscription of Thinger.io Private Cloud Solution. Check [this link](https://thinger.io/pricing) to create your own private instance within minutes.
+2. Install [{{% tts %}} Plugin](https://docs.thinger.io/plugins/the-things-stack) in the instance.
+
+## Plugin Configuration
+
+This section describes the different interfaces that can be used to configure the [{{% tts %}} Plugin](https://docs.thinger.io/plugins/the-things-stack).
+
+### Integrating {{% tts %}} Applications
+
+The first step to perform the integration, is to create a new plugin configuration. It is possible to create multiple configuration profiles with custom behavior for each application deployed in {{% tts %}}. To create a new application profile, just type the Application ID and press the green **Add Application** button. Note that the ID must be exactly the same identifier defined in {{% tts %}} application.
```
Comment (nejraselimovic) [inline doc/content/integrations/cloud-integrations/thinger-io/_index.md:29]:
```suggestion
The first step to perform the integration, is to create a new plugin configuration on Thinger.io Cloud. It is possible to create multiple configuration profiles with custom behavior for each application deployed in {{% tts %}}. To create a new application profile, just type the **Application Id** and press the green **Add Application** button. Note that this ID must exactly match the application ID defined in your {{% tts %}} application.
```

File: doc/content/integrations/cloud-integrations/thinger-io/_index.md [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+{{< figure src="thingerio_default_settings.png" alt="[Thinger.io](https://thinger.io/) {{% tts %}} Plugin default settings" >}}
+
+The `Application Id` dropdown allows to select and configure a particular application profile, but if the "default" profile is selected, the configuration will be applied to all the applications integrated with the plugin.
```
Comment (nejraselimovic) [inline doc/content/integrations/cloud-integrations/thinger-io/_index.md:33]:
```suggestion
The `Application Id` dropdown allows to select and configure a particular application profile, but if the `Default` profile is selected, the configuration will be applied to all the applications integrated with the plugin.
```

File: doc/content/integrations/cloud-integrations/thinger-io/_index.md [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+
+{{< /tabs/tab >}}
+
+{{< /tabs/container >}}
+
+## {{% tts %}} Console Configuration
+
+### Uplink Configuration
+
+The last tab of the plugin configuration interface is called `Webhook Settings`, it has been created to help the developers to complete the integration in {{% tts %}} Console, by providing all the information required to setup the webhook profile.
```
Comment (nejraselimovic) [inline doc/content/integrations/cloud-integrations/thinger-io/_index.md:135]:
```suggestion
The last tab of the plugin configuration interface is called **Webhook Settings**, it has been created to help the developers to complete the integration in {{% tts %}} Console, by providing all the information required to setup the webhook profile. 
```

File: doc/content/integrations/cloud-integrations/thinger-io/_index.md [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+{{< figure src="thingerio_webhook_settings.png" alt="[Thinger.io](https://thinger.io/) {{% tts %}} Plugin webhook settings tab" >}}
+
+{{< note >}}
+The REST API does not define the application ID, this parameter will be checked by the plugin software to manage the payload according to the configuration.
+{{</ note >}}
+
+To create a new webhook integration follow the next steps in {{% tts %}} web console:
+
+1. Select the Application to be integrated.
+2. In the main menu open the `Integrations` section and click the `Webhooks` option. The webhooks list will be shown.
```
Comment (nejraselimovic) [inline doc/content/integrations/cloud-integrations/thinger-io/_index.md:146]:
```suggestion
2. In the main menu open the **Integrations** section and click the **Webhooks** option. The webhooks list will be shown.
```

File: doc/content/integrations/cloud-integrations/thinger-io/_index.md [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+
+{{< figure src="thingerio_tts_sidebar.png" alt="{{% tts %}} sidebar" >}}
+
+3. Clicking the **+Add webhook** blue button in the right top corner of the interface allows choosing between different webhooks integration templates. Select [Thinger.io](https://thinger.io/) template. Then, configure the webhook only requires filling the form with the information provided by [Thinger.io](https://thinger.io/) `Webhook Settings` tab and selecting JSON webhook format.
```
Comment (nejraselimovic) [inline doc/content/integrations/cloud-integrations/thinger-io/_index.md:150]:
```suggestion
3. Clicking the **+Add webhook** blue button in the right top corner of the interface allows choosing between different webhooks integration templates. Select [Thinger.io](https://thinger.io/) template. Then, configure the webhook only requires filling the form with the information provided by [Thinger.io](https://thinger.io/) **Webhook Settings** tab and selecting JSON webhook format.
```

File: doc/content/integrations/cloud-integrations/thinger-io/_index.md [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+
+{{< figure src="thingerio_tts_webhook.png" alt="[Thinger.io](https://thinger.io/) {{% tts %}} webhook settings" >}}
+
+{{< note >}}
+The Authorization header must be set up using the access token including "Bearer" command
```
Comment (nejraselimovic) [inline doc/content/integrations/cloud-integrations/thinger-io/_index.md:155]:
```suggestion
The **Authorization Header** must be set up using the access token including `Bearer` command.
```

Comment (nejraselimovic) [reply doc/content/integrations/cloud-integrations/thinger-io/_index.md:155]: You can best spot the difference of using **this** and `this` on this example. For buttons, tabs, and field names in the Console we use **this**, while for variable values just like this one we use `this`. This is why I added some of these suggestions in the rest of the doc.

**Review by nejraselimovic — APPROVED** [review]
@bautisflow I'm going to approve this since this documentation is great :) but before merging, I added a few suggestions for minor style improvements. Maybe it would also be good to add a screenshot at the end so that users can see how it looks on Thinger.io when messages arrive or some kind of data chart?

**Comment by bautisflow** [comment]
Thanks for your suggestions @nejraselimovic . I've applied them as well as added a Monitor Your Data section with a couple of additional screenshots.
monitor_your_data

**Comment by nejraselimovic** [comment]
> Thanks for your suggestions @nejraselimovic . I've applied them as well as added a Monitor Your Data section with a couple of additional screenshots. monitor_your_data

Great!

**Pull request merged**